### PR TITLE
Post comment

### PR DESCRIPTION
**Pull request by captainread**
postComment 201, 
400s for missing fields and empty strings, 
404s for invalid review IDs and usernames (was the latter needed? assuming want behaviour of "registered users" commenting)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -202,4 +202,84 @@ describe("/api/reviews/:review_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send(testCommentGhost)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404: Not Found");
+      });
+  });
+
```
Comment (emilyb93) [inline __tests__/app.test.js]: Brilliant testing here, would just like to see one more if the review id is not a number `/api/reviews/twister/comments`

File: controllers/reviews.controller.js [hunk 2]
```diff
@@ -29,3 +30,14 @@ exports.getCommentsByReviewID = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.postComment = (req, res, next) => {
+  review_id = req.params.review_id;
+  newComment = req.body;
```
Comment (emilyb93) [inline controllers/reviews.controller.js]: VARIABLES WITHOUT DECLARATIONS?! straight to jail :/

File: models/reviews.model.js [hunk 3]
```diff
@@ -43,3 +43,23 @@ exports.selectCommentsByReviewID = (review_id) => {
...
+  ) {
+    return Promise.reject({ status: 400, msg: "400: Bad Request" });
+  } else {
+    return checkExists("reviews", "review_id", review_id).then(() => {
+      return checkExists("users", "username", newComment.username).then(() => {
+        let queryStr = `INSERT INTO comments (review_id, author, body) VALUES ($1, $2, $3) RETURNING *;`;
+        const queryVals = [review_id, newComment.username, newComment.body];
+        return db.query(queryStr, queryVals).then((result) => {
+          return result.rows[0];
+        });
```
Comment (emilyb93) [inline models/reviews.model.js]: nested promises, tut tut. Make sure we are only dealing with one promise chain please

**Review by emilyb93 — CHANGES_REQUESTED** [review]
couple of changes but good work!

File: controllers/reviews.controller.js (hunk 2 above)
Comment (captainread) [reply controllers/reviews.controller.js]: 🚨🚔👮⚖️🧑‍⚖️⛓️😭

**Pull request merged**